js/content/community.js

```diff
@@ -3670,42 +3671,75 @@ let WorkshopBrowseClass = (function(){
-                });
+                let formData = new FormData();
+                formData.append("sessionid", User.getSessionId());
+                formData.append("appid", appid);
+                formData.append("id", id);
+
+                return RequestData.post("https://steamcommunity.com/sharedfiles/" + method, formData, {
+                    withCredentials: true
+                }, true)
+                .then(function(res) {
+                    if (!res || !res.success || res.success !== 1) {
+                        failed++;
+                        console.error(new Error("Bad response"))
+                    }
+                    completed++;
+                    updateWaitDialog();
+                })
+                .catch(function(err) {
+                    failed++;
+                    completed++;
+                    updateWaitDialog();
+                    console.error(err);
+                })
             }
 
             Messenger.addMessageListener("startSubscriber", async function() {
                 updateWaitDialog();
 
+                function canSkip(method, node) {
+                    if (method === "subscribe") {
+                        return node && node.style.display !== "none";
+                    }
+
+                    if (method === "unsubscribe") {
+                        return !node || node.style.display === "none";
+                    }
+
+                    return false;
+                }
+
                 let workshopItems = [];
                 for (let p = 1; p <= Math.ceil(total / 30); p++) {
                     let url = new URL(window.location.href);
                     url.searchParams.set("p", p);
                     url.searchParams.set("numperpage", 30);
     
-                    let result = await RequestData.getHttp(url.toString());
-                    let xmlDoc = new DOMParser().parseFromString(result, "text/html");
+                    let result = await RequestData.getHttp(url.toString()).catch(err => console.error(err));
+                    if (!result) {
+                        await sleep(10000);
+                        p--;
+                        continue;
+                    }
 
-                    for (let node of xmlDoc.querySelectorAll(".workshopItemPreviewHolder")) {
+                    let xmlDoc = new DOMParser().parseFromString(result, "text/html");
+                    for (let node of xmlDoc.querySelectorAll(".workshopItem")) {
+                        let subNode = node.querySelector(".user_action_history_icon.subscribed");
+                        if (canSkip(method, subNode)) { continue; }
+                    
+                        node = node.querySelector(".workshopItemPreviewHolder");
                         workshopItems.push(node.id.replace("sharedfile_", ""))
                     }
                 }
+
+                total = workshopItems.length;
+                updateWaitDialog();
     
                 Promise.all(
```

Comment:
I know that this isn't part of this PR, but the formatting of this statement suggest that `finally` is called on the result of `Array.map`, although it is called on `Promise.all`

Reply:
This formatting was done by @tfedor. I suggest you discuss this with him.